tidy3d/plugins/design/method.py

```diff
@@ -23,335 +20,780 @@ class Method(Tidy3dBaseModel, ABC):
     )
 
-    @abstractmethod
-    def get_sampler(self, parameters: Tuple[ParameterType, ...]) -> qmc.QMCEngine:
-        """Sampler for this ``Method`` class. If ``None``, sets a default."""
+    def any_to_int_param(self, parameter):
+        """Convert ParameterAny object to integers and provide a conversion dict to return"""
 
-    def sample(self, parameters: Tuple[ParameterType, ...], **kwargs) -> Dict[str, Any]:
-        """Defines how the design parameters are sampled on grid."""
+        return dict(enumerate(parameter.allowed_values))
 
-        sampler = self.get_sampler(parameters)
-        pts_01 = sampler.random(self.num_points)
+    def sol_array_to_dict(
+        self, solution: np.array, keys: list, param_converter: dict
+    ) -> list[dict]:
+        """Convert an array of solutions to a list of dicts for function input"""
+        sol_dict_list = [dict(zip(keys, sol)) for sol in solution]
 
-        # for each dimension, sample `num_points` points and combine them all
-        result = {}
-        for i, design_var in enumerate(parameters):
-            pts_i_01 = pts_01[..., i]
-            values = design_var.select_from_01(pts_i_01)
-            result[design_var.name] = values
-        return result
+        self._handle_param_convert(param_converter, sol_dict_list)
 
+        return sol_dict_list
 
-class MethodMonteCarlo(AbstractMethodRandom):
-    """Select sampling points using Monte Carlo sampling (Latin Hypercube method).
+    def _handle_param_convert(self, param_converter: dict, sol_dict_list: list[dict]) -> None:
+        for param, convert in param_converter.items():
+            for sol in sol_dict_list:
+                if isinstance(sol[param], float):
+                    sol[param] = int(round(sol[param], 0))
+                sol[param] = convert[sol[param]]
 
-    Example
-    -------
-    >>> import tidy3d.plugins.design as tdd
-    >>> method = tdd.MethodMonteCarlo(num_points=20)
+
+class MethodBayOpt(MethodOptimize, ABC):
+    """A standard method for performing a bayesian optimization search.
+
+    The fitness function is maximising by default.
     """
 
-    def get_sampler(self, parameters: Tuple[ParameterType, ...]) -> qmc.QMCEngine:
-        """Sampler for this ``Method`` class."""
+    initial_iter: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of initial random search iterations",
+        description="The number of search runs to be done initialially with parameter values picked randomly. This provides a starting point for the Gaussian processor to optimize from. These solutions can be computed as a single ``Batch`` if the pre function generates ``Simulation`` objects.",
+    )
 
-        d = len(parameters)
-        return DEFAULT_MONTE_CARLO_SAMPLER_TYPE(d=d)
+    n_iter: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of bayesian optimization iterations",
```

Comment:
Title case, and I noticed from time to time `bayesian` not being capitalized. In some outputs, just FYI